JOSESwift/Sources/CryptoImplementation/DataTimingSafeCompare.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  DataTimingSafeCompare.swift
+//  JOSESwift
+//
+//  Created by Tobias Hagemann on 14.04.21.
+//
+
+import Foundation
+
+extension Data {
+    /// Compares data in constant-time.
+    ///
+    /// The running time of this method is independent of the data compared, making it safe to use for comparing secret values such as cryptographic MACs.
+    ///
+    /// The number of bytes of both data are expected to be of same length.
+    ///
+    /// - Parameter other: Other data for comparison.
+    /// - Returns: `true` if both data are equal, otherwise `false`.
+    func timingSafeCompare(with other: Data) -> Bool {
+        assert(self.count == other.count, "parameters should be of same length")
+        if #available(iOS 10.1, *) {
```

Comment:
what about [macOS](https://developer.apple.com/documentation/kernel/3197718-timingsafe_bcmp) and tvOS?

Reply:
According to the `Package.swift` file, this library requires macOS 10.15 or higher, which definitely supports `timingsafe_bcmp()`. This library doesn't support tvOS, as far as I can tell, that's why only iOS 10.1 needs to be checked (because this library requires iOS 10.0 or higher).

Reply:
In that case, I'd suggest using `timingsafe_bcmp` on macOS as well. Rather than the loop.

Reply:
Should already be the case. That's what the `*` stands for.

Reply:
I'm not sure if I can find any documentation on the web but if you navigate "inside" the function, you'll land in `usr/include/string.h`, which has the compatibility annotations:

```
__OSX_AVAILABLE(10.12.1) __IOS_AVAILABLE(10.1)
__TVOS_AVAILABLE(10.0.1) __WATCHOS_AVAILABLE(3.1)
int	timingsafe_bcmp(const void *__b1, const void *__b2, size_t __len);
```

Weirdly enough, it's probably even available on macOS earlier than 10.15. 🤷

Reply:
Oh yeah good point. 
Interesting that they have different availability in the docs vs. in code but I guess the code annotation is good enough for us. ✅